### PR TITLE
Testing : update to 0.12.0, nginx fix, other fixes

### PR DESCRIPTION
**Pull request by lapineige**
#57 : Updated to V0.12.0 and Nginx fix
#59

[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/shaarli_ynh%20PR58%20(ericgaspar)/)

### CODE REVIEW

File: issue_template.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+- Are you in a special context or did you perform some particular tweaking on your YunoHost instance ?: *no / yes*
+  - If yes, please explain:
+- Using, or trying to install package version/branch:
+- If upgrading, current package version: *can be found in the admin, or with `yunohost app info $app_id`*
+
+### Steps to reproduce
+
+- *If you performed a command from the CLI, the command itself is enough. For example:*
+    ```sh
+    sudo yunohost app install REPLACEBYYOURAPP
```
Comment (ericgaspar) [inline issue_template.md:34]:
```suggestion
    sudo yunohost app install shaarli
```

File: scripts/_common.sh [hunk 2]
```diff
@@ -1,95 +1,30 @@
...
 #=================================================
 
 # dependencies used by the app
-pkg_dependencies="php-cli php-gettext php-curl php-intl php-gd php-mbstring openssl"
+YNH_PHP_VERSION="7.3"
+# dependencies used by the app
+# dependencies used by the app
+pkg_dependencies=""
+
+extra_php_dependencies="php${YNH_PHP_VERSION}-cli php${YNH_PHP_VERSION}-gettext php${YNH_PHP_VERSION}-curl php${YNH_PHP_VERSION}-intl php${YNH_PHP_VERSION}-gd php${YNH_PHP_VERSION}-mbstring openssl"
```
Comment (ericgaspar) [inline scripts/_common.sh:13]:
```suggestion
YNH_PHP_VERSION="7.3"

# dependencies used by the app
pkg_dependencies="openssl"

extra_php_dependencies="php${YNH_PHP_VERSION}-cli php${YNH_PHP_VERSION}-gettext php${YNH_PHP_VERSION}-curl php${YNH_PHP_VERSION}-intl php${YNH_PHP_VERSION}-gd php${YNH_PHP_VERSION}-mbstring"
```
`openssl` in not a PHP package so better to install it with `pkg_dependencies`

**Review by ericgaspar — CHANGES_REQUESTED** [review]
Quick review...
You can also remove the comments on install script...

**Comment by anmol26s** [comment]
Why is  CI failing?

**Review by alexAubin — APPROVED** [review]
Super quick review, LGTM

**Comment by lapineige** [comment]
@anmol26s I don't get it https://ci-apps-dev.yunohost.org/jenkins/job/shaarli_ynh%20PR58%20(ericgaspar)/lastBuild/console

**Comment by alexAubin** [comment]
@lapineige : uh yeah for some reason it looks like the CI fails to reach the app after the (successful) installation:

```
Try to access by url...
curl: (47) Maximum (50) redirects followed
```

Dunno if that's due to the CI somehow or if an human is able to reproduce the issue...

**Comment by ericgaspar** [comment]
I think the app is redirected (to a first logging post-install panel) and NGINX is not configured to redirect.

You can implement the cURL post-install explained by Kayou on this issue for Mantis:
https://github.com/YunoHost-Apps/mantis_ynh/pull/6
and the code:
https://github.com/YunoHost-Apps/mantis_ynh/blob/e7a1f471d813a829558fe426aa21beb124f76727/scripts/install#L121-L137
This will skip the post installation panel (and also the 302 on NGINX)

Another solution could be perhaps to add a proxy_redirect in the NGINX configuration ...

**Comment by lapineige** [comment]
I don't understand all of that, does it concerns only CI or this app too ?

**Comment by ericgaspar** [comment]
I think this is a problem with the NGINX config ...

**Comment by ericgaspar** [comment]
@lapineige I pushed Patch2 which should fix some errors

**Comment by tituspijean** [comment]
So, I have investigated the issue with the `Too many redirects`. It happens due to Shaarli making the browser go back and forth between `/install` and `/install/session-test`. (I suspect it is a feature, not a bug, but I fail to see *why* they check for a session that way).

If we reinstate `conf/config.json.php`, the postinstall will be ignored and redirects loop will never be triggered. We need to update this file though, and put the placeholders `sed`s (or `--other_vars`) back in `install` and `upgrade`.

**Comment by lapineige** [comment]
I'll trust you on this one, this is beyond my knowledge in yunohost packaging and nginx stuff

**Comment by tituspijean** [comment]
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/shaarli_ynh%20PR58%20(tituspijean)/) as of 9a949c1.

**Review by alexAubin — APPROVED** [review]
Super quick review, LGTM

**Review by ericgaspar — APPROVED** [review]

**Pull request merged**